### PR TITLE
call confirm popover with times remaining

### PR DESCRIPTION
**Pull request by wittejm**
Requires this api change: https://github.com/Ameelio/connect-api-nest/pull/681

https://user-images.githubusercontent.com/2104990/151882440-b9be063b-546a-49f6-85d4-551c8908354d.mov

### CODE REVIEW

File: src/components/Inbox/VoiceCall/StartVoiceCallPopover.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+const StartVoiceCallPopover: React.FC<Props> = ({
+  onConfirm,
+  onClose,
+  isOpen,
+  children,
+}) => {
+  const { data } = useQuery(GetVoiceCallTimesDocument, {
+    onError: (err) =>
+      reportApolloError(err, "StartVoiceCallPopover/GetVoiceCallTimesDocument"),
```
Comment (cainlevy) [inline src/components/Inbox/VoiceCall/StartVoiceCallPopover.tsx:20]: slight improvement from earlier today:

```suggestion
    onError: errorReporter("StartVoiceCallPopover/GetVoiceCallTimesDocument"),
```

File: src/components/Inbox/VoiceCall/StartVoiceCallPopover.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  });
```
Comment (cainlevy) [inline src/components/Inbox/VoiceCall/StartVoiceCallPopover.tsx:21]: should we customize the network policy for this query to manage caching?

on the one hand, it seems like time sensitive data that could get stale quickly. on the other hand, we don't (currently) expect pathways clients to remain logged in for long.

File: src/utils/Time.ts [hunk 3]
```diff
@@ -17,3 +17,13 @@ export function intervalSecondsToMinutesLabel(seconds?: number | null): string {
...
 }
+
+export function millisecondsToMinutesAndSeconds(
+  milliseconds: number | undefined
+): string {
+  if (!milliseconds) return "";
+  const minutes = Math.floor(milliseconds / (60 * 1000));
+  const remainder = milliseconds - minutes * 60 * 1000;
+  const seconds = `${Math.floor(remainder / 1000)}`.padStart(2, "0");
+  return `${minutes}:${seconds}`;
```
Comment (cainlevy) [inline src/utils/Time.ts:28]: non-blocking UX thought: `seconds` may be a distracting and unhelpful amount of granularity

**Review by cainlevy — APPROVED** [review]

File: src/components/Inbox/VoiceCall/StartVoiceCallPopover.tsx (hunk 2 above)
Comment (wittejm) [reply src/components/Inbox/VoiceCall/StartVoiceCallPopover.tsx]: I considered using `refetch` somewhere but didn't see a clever place to put it.

Then I started writing this comment above and thought of a clever place to put it, so, thanks!

File: src/utils/Time.ts (hunk 3 above)
Comment (wittejm) [reply src/utils/Time.ts:28]: Makes sense! The ticket is explicitly calling it low-fi so I'll leave this as is.

File: src/components/Inbox/VoiceCall/StartVoiceCallPopover.tsx (hunk 1 above)
Comment (wittejm) [reply src/components/Inbox/VoiceCall/StartVoiceCallPopover.tsx:20]: I see that's in DOC client, not Pathways yet.

Comment (cainlevy) [reply src/components/Inbox/VoiceCall/StartVoiceCallPopover.tsx:20]: yep my bad!

**Pull request merged**